Panic in Storage API

Getting a panic when reading data from the storage API. Here is the query:

```
select(db:"telegraf").range(start:-240h).sum()
```


```
panic: unreachable: *storage.stringMultiShardBatchCursor

goroutine 2839 [running]:
github.com/influxdata/influxdb/services/storage.newSumBatchCursor(0x10dc260, 0xc42deae770, 0x0, 0x16)
        /home/nathanielc/projects/influxdb/src/github.com/influxdata/influxdb/services/storage/batch_cursor.go:43 +0x34f
github.com/influxdata/influxdb/services/storage.newAggregateBatchCursor(0x10de4e0, 0xc420016150, 0xc42a0e1948, 0x10dc260, 0xc42deae770, 0xb5, 0xc4330e1b80)
        /home/nathanielc/projects/influxdb/src/github.com/influxdata/influxdb/services/storage/batch_cursor.go:25 +0x80
github.com/influxdata/influxdb/services/storage.(*ResultSet).Cursor(0xc4251a0240, 0x1, 0x0)
        /home/nathanielc/projects/influxdb/src/github.com/influxdata/influxdb/services/storage/resultset.go:43 +0xfb
github.com/influxdata/influxdb/services/storage.(*rpcService).Read(0xc4331cad80, 0xc42d58c700, 0x10deae0, 0xc436734c80, 0xc4331cad80, 0xc424b45598)
        /home/nathanielc/projects/influxdb/src/github.com/influxdata/influxdb/services/storage/rpc_service.go:84 +0x19c
github.com/influxdata/influxdb/services/storage._Storage_Read_Handler(0xbe05a0, 0xc4331cad80, 0x10dcda0, 0xc424b45580, 0xc430dd5779, 0xc430e48248)
        /home/nathanielc/projects/influxdb/src/github.com/influxdata/influxdb/services/storage/storage.yarpc.go:134 +0x10e
github.com/influxdata/yarpc.(*Server).handleStreamingRPC(0xc4331b8210, 0xc436c90000, 0xc4331cada0, 0x1114500)
        /home/nathanielc/projects/influxdb/src/github.com/influxdata/yarpc/server.go:220 +0xfe
github.com/influxdata/yarpc.(*Server).handleStream(0xc4331b8210, 0xc436c90000)
        /home/nathanielc/projects/influxdb/src/github.com/influxdata/yarpc/server.go:199 +0x1c6
created by github.com/influxdata/yarpc.(*Server).serveSession
        /home/nathanielc/projects/influxdb/src/github.com/influxdata/yarpc/server.go:166 +0x4a
```

--- a/services/storage/batch_cursor.go
+++ b/services/storage/batch_cursor.go
@@ -40,7 +40,8 @@ func newSumBatchCursor(cur tsdb.Cursor) tsdb.Cursor {
 	case tsdb.UnsignedBatchCursor:
 		return &unsignedSumBatchCursor{UnsignedBatchCursor: cur}
 	default:
-		panic(fmt.Sprintf("unreachable: %T", cur))
+		// TODO(sgc): propagate an error instead?
+		return nil
 	}
 }
 

